src/auth/apps.rs

```diff
@@ -0,0 +1,32 @@
+use jsonwebtoken::{encode, Algorithm, EncodingKey, Header};
+use serde::{Deserialize, Serialize};
+use std::time::SystemTime;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Claims<'a> {
+    iss: &'a str,
+    iat: usize,
+    exp: usize,
+}
+
+/// Create a JSON Web Token that can be used to authenticate an a GitHub application.
+///
+/// See: https://docs.github.com/en/developers/apps/getting-started-with-apps/setting-up-your-development-environment-to-create-a-github-app#authenticating-as-a-github-app
+pub fn create_authenticate_as_app_jwt(
+    github_app_id: &str,
+    private_key: &str,
+) -> Result<String, jsonwebtoken::errors::Error> {
+    let key = EncodingKey::from_rsa_pem(private_key.as_bytes())?;
```

Comment:
This shouldn't be stringly typed. `app_id` is a number, and `private_key` is a bag of bytes. Using `A: AsRef<[u8]>` allows us to accept any type which can be referenced as a bag of bytes.

```suggestion
pub fn create_jwt<A: AsRef<[u8]>>(
    github_app_id: AppId,
    private_key: A,
) -> Result<String, jsonwebtoken::errors::Error> {
    let key = EncodingKey::from_rsa_pem(private_key.as_ref())?;
```